diff -Naur backup/play-services-basement/src/main/java/org/microg/gms/common/GmsService.java input/play-services-basement/src/main/java/org/microg/gms/common/GmsService.java
--- backup/play-services-basement/src/main/java/org/microg/gms/common/GmsService.java	2025-03-15 18:17:58.241999358 +0300
+++ input/play-services-basement/src/main/java/org/microg/gms/common/GmsService.java	2025-03-15 18:20:05.681999310 +0300
@@ -16,6 +16,8 @@
 
 package org.microg.gms.common;
 
+import com.google.android.gms.common.BuildConfig;
+
 public enum GmsService {
     UNKNOWN(-2),
     ANY(-1),
